### DISCUSSION TITLE
fpv.py example

### DISCUSSION THREAD

**Discussion by 4perros** [bitcraze/discussions#267] [Q&A]
Hello Marcus
I am trying to run the fpv.py example, I am not sure how to do it. I am running the script from inside the folder where it is located with python3 Is there anything else I need to do before running it.
I get the following output :

INFO:cflib.crazyflie:Callback->Connection initialized[radio://0/80/2M/E7E7E7E7E7]
INFO:cflib.crazyflie:We are connected[radio://0/80/2M/E7E7E7E7E7], request connection setup
INFO:cflib.crazyflie.platformservice:Request _request_protocol_version()
Not connecting with WiFi
INFO:cflib.crazyflie:Closing link
INFO:cflib.crazyflie:Callback->Connected to [radio://0/80/2M/E/E7E7E7E7]
INFO:cflib.crazyflie:Callback->Disconnected from [radio://0/80/2M/E7E7E7E7E7]
Disconnected

the radio blinks briefly so does the crazyflie, but then it stops with the above output.

Regards 
Rodrigo

**Comment by evoggy** [comment]
Hi!

The script by default is trying to use the Crazyradio (that's probably a bug). You have to set a different URI (like tcp://x.x.x.x:5000 or tcp://aideck-AABBCCDD.local) for it to connect using the network. This can be done directly in the script or from the command line:
```CFLIB_URI=tcp://x.x.x.x:5000 fpv.py```

**Comment by 4perros** [comment]
hello Marcus
Thank you for your reply. I followed your instructions and the application run and connected to Crazyflie and a small window opened, but nothing else happened, the crazyflie didn’t move, this is the output

INFO:cflib.crazyflie:Callback->Connection.initialized[tcp://192.168.4.1:5000]
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
CPX socket transport
Connecting to socket on 192.168.4.1:5000…
Connected
Creating queue for CPXFunction.CRTP
INFO:cflib.crazyflie:We are connected[tcp://192.168.4.1:5000]. request connection setup
INFO:cflib.crazyflie.platformservice:Request _request_protocol_version()
Creating queue for CPXFunction.APP

does the application uses opencv-python for the window guy? do i use ctrl+c to exit?

regards
Rodrigo

**Reply by evoggy** [reply]
The app uses Qt, so you should get a window which you can just close. But if there's no video coming from the GAP8 the windows will probably be small (since it will expand when rendering the frame). Have you run any other examples on the deck, like the wifi-img-streamer?

**Reply by 4perros** [reply]
hello
i am trying to flash the wifi streamer example using cfloader it starts then when it gets to 6% it stops. the leds turn off so does the radio then almost immediately comes back on but nothing happens, the red led starts blinking very fast, I let it run for a few seconds but nothing changes. Could you let me know what could be the problem.

regards
rodrigo

**Reply by knmcguire** [reply]
@4perros  this is another issue. Please make a new discussion about your flashing problem

**Reply by knmcguire** [reply]
Also we had quite some support questions about it as well so please check the aideck tag in discussions: https://github.com/bitcraze/discussions/discussions?discussions_q=label%3A%22AI+deck%22

**Comment by 4perros** [comment]
Hello
I have resolved the issue. I did a fresh reinstallation now it all works fine, although the Crazyflie doesn’t lift off when I run the fpv.py example.
One more  question, if I flash the Crazyflie with the latest firmware would it affect the aideck settings?
Regards
Rodrigo

**Reply by knmcguire** [reply]
For not taking off, did you also attach a flowdeck?

The last question should be in a new discussion as I explained before.. but I'll make an exception here: If you install the latest release through the cfclient bootloader dialog, yes it will change the aideck settings if you flashed the wifi example via these instructions: https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/test-functions/wifi-streamer/. It gets the wifi name and password or if it needs to connect to a wifi point through the firmware, which needs to be configured with make menuconfig and flashed through make cload.

**Comment by 4perros** [comment]
Hi
Thank you very much for your help. Regarding the not taking off, yes I have a flow deck attached, but I am using an older version (Rev.d) not version 2. Could this be the problem?
I’ll try it again after I upgrade the firmware.
Regards
Rodrigo

**Comment by 4perros** [comment]
hello Marcus
Thank you for your reply. I followed your instructions and the application run and connected to Crazyflie and a small window opened, but nothing else happened, the crazyflie didn’t move, this is the output

INFO:cflib.crazyflie:Callback->Connection.initialized[tcp://192.168.4.1:5000]
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
CPX socket transport
Connecting to socket on 192.168.4.1:5000…
Connected
Creating queue for CPXFunction.CRTP
INFO:cflib.crazyflie:We are connected[tcp://192.168.4.1:5000]. request connection setup
INFO:cflib.crazyflie.platformservice:Request _request_protocol_version()
Creating queue for CPXFunction.APP

does the application uses opencv-python for the window guy? do i use ctrl+c to exit?

regards
Rodrigo

**Reply by evoggy** [reply]
The fpv.py example doesn't use opencv, it only uses qt. If there's no window being shown then you're probably not getting any data from the Crazyflie. Can you try to restart the Crazyflie and try again? Have you tested the GAP8/ESP32 firmware with the streamer example and gotten this to work?

**Comment by 4perros** [comment]
hi
I have run the wifi streamer example many times, but after i installed the crazyflie-lib-python that comes with examples and tried to run the fpv.py and nothing happened, i tried to run the streamer again and wouldn’t work, so i uninstalled and reinstalled opencv-python several times, but still cannot fix it.
I think this an issue with opencv-python. I noticed that i had opencv-python and opencv-python-headless installed and the streamer  example worked, but after installing lib-python it stopped so i uninstalled opencv-python-headless and that’s when it stopped. But so far nothing has helped.

regards
rodrigo

**Reply by evoggy** [reply]
Yes, there's going to be a problem if you have both installed but not the same version. But opencv isn't used in the fpv.py example, only in the streamer example. If you're having issues with the streamer example please start a new discussion about this so we keep this separate from the fpv.py issues.
